pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/protobuf/CodecParseMethodGenerator.java

```diff
@@ -303,6 +316,46 @@ private static void generateFieldCaseStatement(final StringBuilder sb, final Fie
                     .replace("$maxSize", String.valueOf(field.maxSize()))
                     .indent(DEFAULT_INDENT)
             );
+        } else if (field.type() == Field.FieldType.MAP) {
+            // This is almost like reading a message above because that's how Protobuf encodes map entries.
+            // However(!), we read the key and value fields explicitly to avoid creating temporary entry objects.
+            final MapField mapField = (MapField) field;
+            final List<Field> mapEntryFields = List.of(mapField.keyField(), mapField.valueField());
+            sb.append("""
+						final var __map_messageLength = input.readVarInt(false);
```

Comment:
Here you seem to use a double underscore prefix. `__` to variable names. Would nice to be consistent.

Reply:
Yes, I do, and I do so by design. This particular change is to support parsing maps with values of type message.

Maps in protobuf are serialized as repeated entry messages, so I'm using a code that is very similar to parsing a message (but not quite the same) in order to parse the map - it's the code we're looking at now. However, when the map value type is a message, this code generator would insert a code snippet related to parsing that message inside of this map parsing code, and this will introduce an inner code in this parsing loop that would again use a variable name `messageLength`, which javac wouldn't allow us to do.

So I'm adding a prefix to the map-related variables and making them look distinct enough so that a human reader of the generated code can quickly spot what `messageLength` the code is dealing with. I thought a double underscore prefix is distinct enough. (I also use triple underscore prefix for map synthetic field names, btw, and again, I do so by design - to try and avoid any conflicts with regular field names.)

Please let me know if you're strongly opposed to multiple underscores and if so, please provide suggestions on how you'd want this to look instead.